### ISSUE TITLE
Web - App triggers Pay button for request money on selection of emoji

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23804]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app and login with user A
2. Open user B report
3. From another device, send request money request from user B to user A
4. Open emoji picker from user A device and search for any emoji
5. Press enter to select the emoji and observe that along with emoji selection, app also approves the request money request

## Expected Result:
 App should not approve request money on selection of emoji from emoji picker

## Actual Result:
App approves request money on selection of emoji from emoji picker

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.47-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/dab1ddd5-3226-46ad-a38a-eaf3d57a8cdd

https://github.com/Expensify/App/assets/93399543/6a1ab0e3-8a46-44aa-a719-fd36b180f263

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690539769279639

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
Only reproduced with public account, not expensifail

**Comment by muttmuure** [comment]
Wow this is really weird

**Comment by muttmuure** [comment]
Can't reproduce this one.

**Comment by dhanashree-sawant** [comment]
Hi @muttmuure, I could still reproduce on latest staging, can you let me know what do you see after trying to recreate? maybe I can help

**Comment by luacmartins** [comment]
This is still happening and can be easily reproducible using a public domain account, e.g. `@gmail.com`

**Comment by muttmuure** [comment]
I wasn't able to reproduce it myself

**Comment by muttmuure** [comment]
Seems kinda sporadic

**Comment by huytran46** [comment]
Hi team, new contributor here.

Upon some debugging, I noticed that the `avoidKeyboard` property is used in the EmojiPicker popover props, but it appears to be ignored within the `PopoverProvider` context, allowing keydown events to propagate.
<img width="814" alt="Screenshot 2023-09-04 at 23 45 44" src="https://github.com/Expensify/App/assets/39490493/477008e8-1a92-4e9f-8240-bd19bc0565be">

After further investigation, it seems that the `ButtonWithDropdownMenu` component which is the `Pay elsewhere` button, specifically in the context of money actions, listens for the Enter keydown event, which appears to be contributing to this bug.

<img width="1381" alt="Screenshot 2023-09-04 at 23 57 27" src="https://github.com/Expensify/App/assets/39490493/ad736032-8101-4f34-b2d4-0f8a551da73c">

<img width="757" alt="Screenshot 2023-09-04 at 23 55 14" src="https://github.com/Expensify/App/assets/39490493/a71cf792-1928-47b6-898c-28ccc8ebeab4">

I'd like to suggest a solution to address this issue:

Ensure that keydown events propagation is set to true whenever `props.avoidKeyboard` is `true`

I'm eager to collaborate on resolving this bug and contributing to the project's continued success. Please let me know your thoughts and how I can assist further.

**Comment by joh42** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When we press enter to select an emoji, the enter event triggers the pay button for money requests.
### What is the root cause of that problem?
We do not stop enter events originating in the emoji selector from propagating.
### What changes do you think we should make in order to solve the problem?

I suggest we simply stop enter events from propagating here:
https://github.com/Expensify/App/blob/d2bdaece9151c8c65d3b4bc4c9578be663f1950d/src/components/EmojiPicker/EmojiPickerMenu/index.js#L192-L200

Using `keyBoardEvent.stopPropagation();`

Result:

https://github.com/Expensify/App/assets/138504087/c08216aa-73d3-4245-9d15-005b8774ba3d



### What alternative solutions did you explore? (Optional)

**Comment by lukemorawski** [comment]
@Ollyws It's Lucas from Callstack. I'm able to reproduce that problem and happy to have a go on that

**Comment by joh42** [comment]
@Ollyws looking into it the code in that statement is only run if an emoji is highlighted. If we want to catch all enter events regardless of if an emoji is selected, I suggest we add a new statement:
```
            if (!isEnterWhileComposition(keyBoardEvent) && keyBoardEvent.key === 'Enter') {
                keyBoardEvent.stopPropagation();
            }
```
I've updated my proposal with this in mind

**Comment by luacmartins** [comment]
I agree that disabling `pressOnEnter` is better given that we're paying someone and we want to be sure that the action is intended

**Comment by dhanashree-sawant** [comment]
Hi @luacmartins, the issue was raised in july but melvin bot has provided me with 50$ offer, can you check once you are available?

**Comment by luacmartins** [comment]
@muttmuure could you please review the offer sent to @dhanashree-sawant?

**Comment by luacmartins** [comment]
@Ollyws could you please complete the BZ checklist above?

**Comment by muttmuure** [comment]
Offers updated

**Comment by muttmuure** [comment]
Just waiting for everyone to accept

**Comment by dhanashree-sawant** [comment]
Thanks @muttmuure, I have accepted the offer.

**Comment by muttmuure** [comment]
It was created before August 30th so the offers are $1000

**Comment by muttmuure** [comment]
Paid @huzaifa-99 and @dhanashree-sawant

**Comment by muttmuure** [comment]
Just waiting for @Ollyws

**Comment by luacmartins** [comment]
Still working through payments

**Comment by muttmuure** [comment]
Looks like this is eligible for bonuses after all, I will apply those now

**Comment by muttmuure** [comment]
bonuses paid out

**Comment by luacmartins** [comment]
Sounds good to me

**Comment by muttmuure** [comment]
https://github.com/Expensify/Expensify/issues/324583 regression test requested
